nixos/modules/virtualisation/virtualbox-guest.nix

```diff
@@ -67,27 +116,39 @@ in
         # Allow systemd dependencies on vboxguest.
         SUBSYSTEM=="misc", KERNEL=="vboxguest", TAG+="systemd"
       '';
-  } (mkIf cfg.x11 {
-    services.xserver.videoDrivers = [ "vmware" "virtualbox" "modesetting" ];
-
-    services.xserver.config =
-      ''
-        Section "InputDevice"
-          Identifier "VBoxMouse"
-          Driver "vboxmouse"
-        EndSection
-      '';
-
-    services.xserver.serverLayoutSection =
-      ''
-        InputDevice "VBoxMouse"
-      '';
-
-    services.xserver.displayManager.sessionCommands =
-      ''
-        PATH=${makeBinPath [ pkgs.gnugrep pkgs.which pkgs.xorg.xorgserver.out ]}:$PATH \
-          ${kernel.virtualboxGuestAdditions}/bin/VBoxClient-all
-      '';
-  })]);
+  }
+    (
+      mkIf cfg.clipboard {
+        systemd.user.services.virtualboxClientClipboard = mkVirtualBoxUserService "--clipboard";
+      }
+    )
+    (
+      mkIf cfg.seamless {
+        systemd.user.services.virtualboxClientSeamless = mkVirtualBoxUserService "--seamless";
+      }
+    )
+    (
+      mkIf cfg.vmsvga {
+        systemd.user.services.virtualboxClientVmsvga = mkVirtualBoxUserService "--vmsvga-session";
+      }
+    )
+    (
+      mkIf cfg.x11 {
+        services.xserver.videoDrivers = [ "vmware" "virtualbox" "modesetting" ];
```

Comment:
Does the DDX actually do anything useful? We should really default to `modesetting` in most cases.

Reply:
Modesetting is also the default, so maybe there is nothing to be done here?

Also the settings below might not be necessary these days.

Reply:
I will try to remove these. They are adopted from the current virtual box guest files in master/stable.